### PR TITLE
Image Customizer: Move fileSystemType to fileSystem type.

### PR DESCRIPTION
**Pull request by cwize1**
Move the `fileSystemType` property from the `partition` type to the `fileSystem` type and rename it to just `type`.

---

###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

###### Does this affect the toolchain?  

NO

###### Test Methodology

- Manually ran image customizer tool.

### CODE REVIEW

**Review by romoh — APPROVED** [review]

File: toolkit/tools/internal/sliceutils/sliceutils.go [hunk 1]
```diff
@@ -119,3 +119,14 @@ func ContainsFunc[K any](inputSlice []K, fn func(K) bool) bool {
 	}
 	return false
 }
+
+func FindValueFunc[K any](inputSlice []K, fn func(K) bool) (K, bool) {
```
Comment (dmcilvaney) [inline toolkit/tools/internal/sliceutils/sliceutils.go:123]: Huh, surprised this doesn't have an implementation in the new slice library... Will be nice to swap some of these out.

Not sure what to put here, this feels a bit wordy but...
```suggestion
FindValueFunc returns the first value in the slice that matches fn(item) == true along with true, otherwise it returns the default value for an instance of value's type, and false.
func FindValueFunc[K any](inputSlice []K, fn func(K) bool) (K, bool) {
```

**Review by PawelWMS — APPROVED** [review]
